build/pluginctl/logs_test.go

```diff
@@ -0,0 +1,202 @@
+package main
+
+import (
+	"fmt"
+	"testing"
+	"time"
+)
+
+func TestCheckOldest(t *testing.T) {
+	for name, tc := range map[string]struct {
+		logs           []string
+		oldest         string
+		expectedLogs   []string
+		expecetdOldest string
+		expectedAllNew bool
+	}{
+		"nil logs": {
+			logs:           nil,
+			oldest:         "oldest",
+			expectedLogs:   nil,
+			expecetdOldest: "oldest",
+			expectedAllNew: false,
+		},
+		"empty logs": {
+			logs:           []string{},
+			oldest:         "oldest",
+			expectedLogs:   nil,
+			expecetdOldest: "oldest",
+			expectedAllNew: false,
+		},
+		"no new entries, one old entry": {
+			logs:           []string{"old"},
+			oldest:         "old",
+			expectedLogs:   []string{},
+			expecetdOldest: "old",
+			expectedAllNew: false,
+		},
+		"no new entries, multipile old entries": {
+			logs:           []string{"old1", "old2", "old3"},
+			oldest:         "old3",
+			expectedLogs:   []string{},
+			expecetdOldest: "old3",
+			expectedAllNew: false,
+		},
+		"one new entry, no old entry": {
+			logs:           []string{"new"},
+			oldest:         "old",
+			expectedLogs:   []string{"new"},
+			expecetdOldest: "new",
+			expectedAllNew: true,
+		},
+		"multipile new entries, no old entry": {
+			logs:           []string{"new1", "new2", "new3"},
+			oldest:         "old",
+			expectedLogs:   []string{"new1", "new2", "new3"},
+			expecetdOldest: "new3",
+			expectedAllNew: true,
+		},
+		"one new entry, one old entry": {
+			logs:           []string{"old", "new"},
+			oldest:         "old",
+			expectedLogs:   []string{"new"},
+			expecetdOldest: "new",
+			expectedAllNew: false,
+		},
+		"one new entry, multipile old entries": {
+			logs:           []string{"old1", "old2", "old3", "new"},
+			oldest:         "old3",
+			expectedLogs:   []string{"new"},
+			expecetdOldest: "new",
+			expectedAllNew: false,
+		},
+		"multipile new entries, ultipile old entries": {
+			logs:           []string{"old1", "old2", "old3", "new1", "new2", "new3"},
+			oldest:         "old3",
+			expectedLogs:   []string{"new1", "new2", "new3"},
+			expecetdOldest: "new3",
+			expectedAllNew: false,
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			logs, oldest, allNew := checkOldest(tc.logs, tc.oldest)
+
+			if allNew != tc.expectedAllNew {
+				t.Logf("expected allNew: %v, got %v", tc.expectedAllNew, allNew)
+				t.Fail()
+			}
```

Comment:
I really wanted to around importing `testify` as that way all plugins would have it in their `go.mod`.